web/controllers/pair_controller.ex

```diff
@@ -63,14 +57,37 @@ defmodule Pairmotron.PairController do
   end
 
   defp generate_pairs(year, week) do
-    User.active_users
+    users = User.active_users
       |> order_by(:id)
       |> Repo.all
+
+    pairs = fetch_pairs(year, week)
+      |> Repo.preload(:users)
+
+    determination = PairBuilder.determify(pairs, users)
+
+    determination.dead_pairs
+      |> Enum.map(fn(p) -> Repo.delete! p end)
+
+    pairs = determination.remaining_pairs
+      |> Repo.preload([:users, :pair_retros])
+      |> Enum.filter(fn(p) -> length(p.pair_retros) == 0 end)
```

Comment:
Tiny optimization here, instead of length == 0 can we do Enum.any? which will short circuit on the first element rather than iterate through all pair_retros. I realize this is a micro-optimization, but I think it might read better too.

`length(p.pair_retros) == 0` vs `Enum.any?(p.pair_retros)`